src/gluonts/model/lstnet/_network.py

```diff
@@ -0,0 +1,341 @@
+        Tensor
+            Shape (batch_size, num_series, 1) if `horizon` was specified
+            and of shape (batch_size, num_series, prediction_length)
+            if `prediction_length` was provided
+            
+        """
+
+        scaled_past_target, _ = self.scaler(
+            past_target.slice_axis(
+                axis=2, begin=-self.context_length, end=None
+            ),
+            past_observed_values.slice_axis(
+                axis=2, begin=-self.context_length, end=None
+            ),
+        )
+        c = self.cnn(scaled_past_target)
+        c = self.dropout(c)
+        c = F.transpose(c, axes=(0, 2, 1))  # NTC
+
+        if F is mx.ndarray:
+            ctx = (
+                c.context
+                if isinstance(c, mx.gluon.tensor_types)
+                else c[0].context
+            )
+            with ctx:
+                rnn_begin_state = self.rnn.begin_state(
+                    func=F.zeros, dtype=self.dtype, batch_size=c.shape[0]
+                )
+        else:
+            rnn_begin_state = self.rnn.begin_state(
+                func=F.zeros, dtype=self.dtype, batch_size=0
+            )
+        r, _ = self.rnn.unroll(
+            inputs=c,
+            length=min(self.conv_out, self.context_length),
+            layout="NTC",
+            merge_outputs=True,
+            begin_state=rnn_begin_state,
+        )
+        r = F.squeeze(
+            F.slice_axis(r, axis=1, begin=-1, end=None), axis=1
+        )  # NC
+        s = self._skip_rnn_layer(F, c)
+        # make fc broadcastable for output
+        fc = self.fc(F.concat(r, s, dim=1)).expand_dims(
+            axis=2
+        )  # N x num_series x 1
+        if self.prediction_length:
+            fc = F.tile(
+                fc, reps=(1, 1, self.prediction_length)
+            )  # N x num_series x prediction_length
+        ar = self._ar_highway(F, past_target)
+        out = fc + ar
+        if self.output_activation is None:
+            return out
+        return (
+            F.sigmoid(out)
+            if self.output_activation == "sigmoid"
+            else F.tanh(out)
+        )
+
+
+class LSTNetTrain(LSTNetBase):
+    @validated()
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.loss_fn = loss.L1Loss()
+
+    def hybrid_forward(
+        self,
+        F,
+        past_target: Tensor,
+        past_observed_values: Tensor,
+        future_target: Tensor,
+    ) -> Tensor:
+        """
+        Computes the training l1 loss for LSTNet for multivariate time-series.
+        All input tensors have NCT layout.
+
+        Parameters
+        ----------
+        F
+        past_target
+            Tensor of shape (batch_size, num_series, context_length)
+        past_observed_values
+            Tensor of shape (batch_size, num_series, context_length)
+        future_target
+            Tensor of shape (batch_size, num_series, 1) if `horizon` was specified
```

Comment:
Is the time length 1 in general when `horizon` is set? In the estimator, the “future length” in the instance splitter is set to `horizon` in this case, so it will be > 1 if `horizon` is. Do I understand this right?

Reply:
Yes, when using `horizon` the since we're predicting a point not a sequence then future is 1 too to compute the loss and for `prediction_length` it's a sequence as usual.

Reply:
I'm expecting instance splitter to respect it, so you're saying it may not?

Reply:
I don’t think so: I would set horizon to 2 and put a breakpoint here to check the shape of future_target, I’m petty sure the time length will be 2, not 1